agreement/cryptoVerifier.go

```diff
@@ -205,7 +205,14 @@ func (c *poolCryptoVerifier) voteFillWorker(toBundleWait chan<- bundleFuture) {
 			}
 
 			uv := votereq.message.UnauthenticatedVote
-			c.voteVerifier.verifyVote(votereq.ctx, c.ledger, uv, votereq.TaskIndex, votereq.message, c.votes.out)
+			err := c.voteVerifier.verifyVote(votereq.ctx, c.ledger, uv, votereq.TaskIndex, votereq.message, c.votes.out)
+			if err != nil && c.votes.out != nil {
+				select {
+				case c.votes.out <- asyncVerifyVoteResponse{index: votereq.TaskIndex, err: err}:
```

Comment:
Looking deeper, this changes the behavior from what happens in AsyncVoteVerifier.executeVoteVerification. Today the only way verifyVote() and EnqueueBacklog() can return errors is if they were canceled, and there is code in `executeVoteVerification` to set cancelled: true in that case. Should this be filled out?

Or maybe no new response should be sent to c.votes.out? It seems like this was the old behavior — there were already cases where verifyVote and EnqueueBacklog might get select-exited, and then verifyVote would just silently drop it, and it would never show up on the cryptoVerifier.VerifiedVotes() channel in demux.

Reply:
Running down what happens when asyncVerifyVoteResponse.cancelled be set — AFAICT all the "if e.Cancelled" code paths result in filtering the event the same way as if e.Err was set.

Reply:
Correct - all the `e.Cancelled` would result in ignoring this event. However, this is not needed in case the reason for exiting is due to a system shutdown, right ?
( i.e. the agreement state machine is going down; it won't be processing this event ).

On the flip side, we could set the Cancelled flag. It wouldn't be wrong, since this is the "difference" between malicious vs. non-malicious vote verification.

Reply:
thinking about that further, I think that you're correct. we should retain the same behavior we have in `executeVoteVerification`.

Reply:
I also just noticed that some of the existing code paths that set err & cancelled also skip req.index (but they all do provide req, which you are omitting here)
```
agreement/asyncVoteVerifier.go
103:            return &asyncVerifyVoteResponse{err: req.ctx.Err(), cancelled: true, req: &req}
112:            return &asyncVerifyVoteResponse{v: v, index: req.index, message: req.message, err: err, cancelled: cancelled, req: &req}
122:            return &asyncVerifyVoteResponse{err: req.ctx.Err(), cancelled: true, req: &req}
130:            return &asyncVerifyVoteResponse{ev: ev, index: req.index, message: req.message, err: err, cancelled: cancelled, req: &req}
`